agent/engine/docker_task_engine_test.go

```diff
@@ -1685,41 +1686,119 @@ func TestUpdateContainerReference(t *testing.T) {
 }
 
 // TestPullAndUpdateContainerReference checks whether a container is added to task engine state when
-// pullSucceeded and DependentContainersPullUpfront is enabled.
+// Test # | Image availability  | DependentContainersPullUpfront | ImagePullBehavior
```

Comment:
Excellent!

Reply:
Thank you!

Reply:
Looks good! Although I think we removed the UTs for when DependentContainersPullUpfront is NOT enabled?

Reply:
`DependentContainersPullUpfrontDisabledWithRemoteImage` is added to TestPullAndUpdateContainerReference. Thank you!